src/m3ninx/index/segment/fst/segment.go

```diff
@@ -158,11 +158,20 @@ func NewSegment(data SegmentData, opts Options) (Segment, error) {
 		numDocs:        metadata.NumDocs,
 		startInclusive: startInclusive,
 		endExclusive:   endExclusive,
-	}, nil
+	}
+
+	// NB(r): The segment uses the context finalization to finalize
```

Comment:
This might be a good place to explain the general approach and how it interacts with outstanding queries

Reply:
Sure yeah.